Show created targets

agustito37 opened the pull request:
## Show created targets

#### Description
As a user I should be able to see all my targets in the map so I can easily identify them

#### Tasks
- The map should be centred on the Intendencia de Montevideo by default
- The map should show all "el centro"
- Given that there are existing targets created in the app, I should be able to see them marked on the map
- I should see a different icon for each topic
- I should see a different radius size depending on the target size

matiasmansilla1989 commented on app/controllers/targets_controller.rb:
```diff
@@ -25,4 +24,9 @@ def create
     params.require(:target).permit(:area, :title, :topic_id, :latitude, :longitude)
   end
+
+  def common_data
```
Controller filters should be used to halt the request cycle and not to load data. 
You can use a simple method instead or a `helper_method` if you want to call them from the view.
Check:
- http://craftingruby.com/posts/2015/05/31/dont-use-before-action-to-load-data.html
- https://apidock.com/rails/ActionController/Helpers/ClassMethods/helper_method

fedeagripa commented on app/assets/javascripts/base/_g_map.js:
```diff
@@ -0,0 +1,121 @@
+var GMap = (() => {
+	var MEDIATOR_NAME = "g_map";
```
Indentation

fedeagripa commented on app/assets/javascripts/base/_g_map.js:
```diff
@@ -0,0 +1,121 @@
+  var _map;
+  var _marker;
+
+	Mediator.installTo(this);
```
same here

fedeagripa commented on app/assets/javascripts/base/_g_map.js:
```diff
@@ -0,0 +1,121 @@
+  }
+
+  function markerChanged(event) {
+    publish(MEDIATOR_NAME+".marker_changed", { lat:event.latLng.lat(), lng:event.latLng.lng() });
```
interpolate the constant so you don't need to concat

fedeagripa commented on app/assets/javascripts/base/_main.js:
```diff
@@ -0,0 +1,18 @@
+  Mediator.installTo(this);
+
+  function init() {
+    publish(MEDIATOR_NAME+".init");
```
Same here about interpolation

fedeagripa commented on app/assets/javascripts/base/_main.js:
```diff
@@ -0,0 +1,18 @@
+  }
+
+  function checkControllerAction(controller, action) {
+    return $("body[data-controller='" + controller + "'][data-action='" + action + "']").length > 0
```
Interpolate so you don't need to switch between single & double quotes

fedeagripa commented on app/assets/javascripts/views/_new_form.js:
```diff
@@ -0,0 +1,57 @@
+var NewForm = (() => {
```
check single or double quotes consinstency and interpolation in the file

fedeagripa commented on app/assets/stylesheets/base/_main.scss:
```diff
@@ -25,7 +25,7 @@ html,body {
 
   h2 {
-    font-size: 12px;
+    font-size: .75rem;
```
If you are switching to rem please do it in all the possible cases (maybe do it in another pr if you wish to)

grilix commented on app/assets/javascripts/base/_g_map.js:
```diff
@@ -0,0 +1,121 @@
+  function init() {
+    createMap();
+    bindEvents();
+    publish(MEDIATOR_NAME+".init",{ lat:DEFAULT_LAT, lng:DEFAULT_LNG });
```
Be careful with the spaces, for example around `+` operator, after `,`, etc. There are a couple of examples where spacing is not consistent.

grilix commented on app/assets/javascripts/base/_g_map.js:
```diff
@@ -0,0 +1,121 @@
+  var MARKER_HEIGHT = 72;
+  var MARKERS_WIDTH = 20;
+  var MARKERS_HEIGHT = 20;
+  var MARKERS_COLOR = '#edcc00';
```
Always use double quotes for strings (unless the standard is to always use single quotes, but don't mix them.)

agustito37 commented:
Removed the before_action sentence in targets controller, added string interpolation in js files so the "+" is not used anymore and fixed indentation problems.
